test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+#include "BoostUnitTest.h"
+
+#include "base/Group.h"
+#include "mechanics/cell/Cell.h"
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/cell/SimpleAssember.h"
+#include "mechanics/constitutive/laws/LinearElastic.h"
+#include "mechanics/constitutive/laws/MechanicsInterface.h"
+#include "mechanics/constraintsPde/Constraints.h"
+#include "mechanics/constraintsPde/ConstraintCompanion.h"
+#include "mechanics/dofs/DofNumbering.h"
+#include "mechanics/integrands/MomentumBalance.h"
+#include "mechanics/integrationtypes/IntegrationTypeTensorProduct.h"
+#include "mechanics/interpolation/InterpolationTrussLinear.h"
+#include "mechanics/mesh/MeshFem.h"
+#include "mechanics/mesh/MeshFemDofConvert.h"
+#include "mechanics/nodes/NodeSimple.h"
+
+#include <cassert>
+#include <functional>
+#include <map>
+#include <vector>
+
+using namespace Eigen;
+using namespace NuTo;
+using namespace NuTo::Groups;
+
+
+constexpr double SpecimenLength = 10.;
+constexpr unsigned int numElements = 4;
+
+//! @brief Standard history data management object.
+//! @tparam T: Data type of the history data
+template <typename T>
+class HistoryData
```

Comment:
IMO this class provides no benefit compared to a `map<int, vector<T>>` written directly at the law. It just provides access and a strange initialization. The main difference may be that it throws a `NuTo::Exception` instead of a `std::out_of_range`.

Reply:
the whole point is to discuss a default "can use but don't need to" class for a standard user, so he does not have to care about history data management (suggested by joergfunger). I know that it is not super, duper nice yet. It is a base for discussion. We can also discuss, if we even wan't to provide such a class.